src/Passwordless/PasswordlessClient.cs

```diff
@@ -24,123 +25,136 @@ public static PasswordlessClient Create(PasswordlessOptions options, IHttpClient
+        _disposeClient = false;
     }
 
-    /// <inheritdoc/>
+    /// <inheritdoc />
     public async Task SetAliasAsync(SetAliasRequest request, CancellationToken cancellationToken)
     {
-        var res = await _client.PostAsJsonAsync("alias",
+        using var response = await _client.PostAsJsonAsync("alias",
             request,
             PasswordlessSerializerContext.Default.SetAliasRequest,
             cancellationToken);
-        res.EnsureSuccessStatusCode();
+
+        response.EnsureSuccessStatusCode();
     }
 
-    /// <inheritdoc/>
+    /// <inheritdoc />
     public async Task<RegisterTokenResponse> CreateRegisterTokenAsync(RegisterOptions registerOptions, CancellationToken cancellationToken = default)
     {
-        var res = await _client.PostAsJsonAsync("register/token",
+        using var response = await _client.PostAsJsonAsync("register/token",
             registerOptions,
             PasswordlessSerializerContext.Default.RegisterOptions,
             cancellationToken);
-        res.EnsureSuccessStatusCode();
-        return (await res.Content.ReadFromJsonAsync<RegisterTokenResponse>(
+
+        response.EnsureSuccessStatusCode();
+
+        return (await response.Content.ReadFromJsonAsync(
             PasswordlessSerializerContext.Default.RegisterTokenResponse,
             cancellationToken))!;
     }
 
-    /// <inheritdoc/>
+    /// <inheritdoc />
     public async Task<VerifiedUser?> VerifyTokenAsync(string verifyToken, CancellationToken cancellationToken = default)
     {
-        var request = new HttpRequestMessage(HttpMethod.Post, "signin/verify")
-        {
-            // TODO: No JsonTypeInfo overload yet?
-            Content = JsonContent.Create(new VerifyTokenRequest(verifyToken)),
-        };
+        using var request = new HttpRequestMessage(HttpMethod.Post, "signin/verify");
+
+        // TODO: No JsonTypeInfo overload yet?
+        request.Content = JsonContent.Create(new VerifyTokenRequest(verifyToken));
 
         // We just want to return null if there is a problem.
         request.SkipErrorHandling();
-        var response = await _client.SendAsync(request, cancellationToken);
+        using var response = await _client.SendAsync(request, cancellationToken);
 
         if (response.IsSuccessStatusCode)
         {
             var res = await response.Content.ReadFromJsonAsync(
                 PasswordlessSerializerContext.Default.VerifiedUser,
                 cancellationToken);
+
             return res;
         }
 
         return null;
     }
 
-    /// <inheritdoc/>
+    /// <inheritdoc />
     public async Task DeleteUserAsync(string userId, CancellationToken cancellationToken = default)
     {
-        await _client.PostAsJsonAsync("users/delete",
+        using var response = await _client.PostAsJsonAsync("users/delete",
             new DeleteUserRequest(userId),
             PasswordlessSerializerContext.Default.DeleteUserRequest,
             cancellationToken);
     }
```

Comment:
Separate issue but: We should probably ensuresuccess here.

Reply:
I was thinking that too but didn't want to change the behavior too much, in case `delete` was meant to be idempotent or something